### ISSUE TITLE
Rank beam_width ctc_beam_search_decoder() results on language model

### ISSUE THREAD

**Issue by kdavis-mozilla** [mozilla/DeepSpeech#408]
Instead of relying on a fix of tensorflow/tensorflow#6034 simply obtain `beam_width` results from `ctc_beam_search_decoder()` and rank them based on the current language model.

This has the advantage not not relying upon the slow Norvig spelling corrector[[1](http://norvig.com/spell-correct.html)]. However, it has the disadvantage of likely spelling worse than including the Norvig spelling corrector[[1](http://norvig.com/spell-correct.html)].

This could should be implemented in a branch, then a WER test should be done comparing it to the current language model + Norvig spelling corrector[[1](http://norvig.com/spell-correct.html)]. Whichever has the lowest WER remains in the main branch.

**Comment by pdcoded** [comment]
Any more updates on this critical piece ?

**Comment by pdcoded** [comment]
@kdavis-mozilla any updates on this issue? Currently the beam search decoder factors in the logit output from acoustic model only and not language model or word weighting if I am not wrong ?

**Comment by reuben** [comment]
I believe this was superceded by the KenLM beam scoring code.
